src/test/lib/MtuTest.cpp

```diff
@@ -0,0 +1,195 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    MsQuic MTU Unittest
+
+--*/
+
+#include "precomp.h"
+#ifdef QUIC_CLOG
+#include "MtuTest.cpp.clog.h"
+#endif
+
+struct MtuTestContext {
+    MsQuicConnection* Connection {nullptr};
+
+    static QUIC_STATUS ConnCallback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT*) {
+        (static_cast<MtuTestContext*>(Context))->Connection = Conn;
+        return QUIC_STATUS_SUCCESS;
+    }
+};
+
+void
+QuicTestMtuSettings()
+{
+    {
+        //
+        // Test setting on library works
+        //
+        MsQuicSettings CurrentSettings;
+        TEST_QUIC_SUCCEEDED(CurrentSettings.GetGlobal());
+
+        MsQuicSettings NewSettings;
+        QUIC_STATUS SetSuccess =
+            NewSettings.
+            SetMinimumMtu(1400).
+            SetMaximumMtu(1400).
+            SetGlobal();
+
+        MsQuicSettings UpdatedSettings;
+        QUIC_STATUS GetSuccess = UpdatedSettings.GetGlobal();
+
+        CurrentSettings.IsSetFlags = 0;
+        CurrentSettings.IsSet.MaximumMtu = TRUE;
+        CurrentSettings.IsSet.MinimumMtu = TRUE;
+        TEST_QUIC_SUCCEEDED(CurrentSettings.SetGlobal());
+
+        TEST_QUIC_SUCCEEDED(SetSuccess);
+        TEST_QUIC_SUCCEEDED(GetSuccess);
```

Comment:
Why do this here? I prefer to eliminate the extra variables and just check for failure at the point of failure.

Reply:
Since its setting the global settings, I want to make sure we successfully revert it back.

Reply:
Ah, I see. You're trying to handle reverting any local changes.